spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java

```diff
@@ -0,0 +1,192 @@
+    Map<String, String> resultMap = new HashMap<>();
+    List<KVDoc> appList = new ArrayList<>();
+    List<KVDoc> serviceList = new ArrayList<>();
+    List<KVDoc> versionList = new ArrayList<>();
+    for (KVDoc kvDoc : resp.getData()) {
+      if (!StringUtils.isEmpty(kvDoc.getStatus()) && !kvDoc.getStatus()
+          .equals(ConfigConstants.STATUS_ENABLED)) {
+        continue;
+      }
+      Map<String, String> labelsMap = kvDoc.getLabels();
+      if (labelsMap.containsKey(ConfigConstants.LABEL_APP) && labelsMap
+          .get(ConfigConstants.LABEL_APP)
+          .equals(serviceCombConfigProperties.getAppName())
+          && labelsMap.containsKey(ConfigConstants.LABEL_ENV) && labelsMap
+          .get(ConfigConstants.LABEL_ENV)
+          .equals(serviceCombConfigProperties.getEnv())) {
+        if (!labelsMap.containsKey(ConfigConstants.LABEL_SERVICE)) {
+          appList.add(kvDoc);
+        }
+        if (labelsMap.containsKey(ConfigConstants.LABEL_SERVICE) && labelsMap
+            .get(ConfigConstants.LABEL_SERVICE)
+            .equals(serviceCombConfigProperties.getServiceName())) {
+          if (!kvDoc.getLabels().containsKey(ConfigConstants.LABEL_VERSION)) {
+            serviceList.add(kvDoc);
+          }
+          if (labelsMap.containsKey(ConfigConstants.LABEL_VERSION) && labelsMap
+              .get(ConfigConstants.LABEL_VERSION)
+              .equals(serviceCombConfigProperties.getServiceName())) {
+            versionList.add(kvDoc);
+          }
+        }
+      }
+    }
+    //kv is priority
+    for (KVDoc kvDoc : appList) {
+      resultMap.putAll(processValueType(kvDoc));
+    }
+    for (KVDoc kvDoc : serviceList) {
+      resultMap.putAll(processValueType(kvDoc));
+    }
+    for (KVDoc kvDoc : versionList) {
+      resultMap.putAll(processValueType(kvDoc));
+    }
+    return resultMap;
+  }
+
+
+  private Map<String, String> processValueType(KVDoc kvDoc) {
+    ValueType vtype;
+    try {
+      vtype = ValueType.valueOf(kvDoc.getValueType());
+    } catch (IllegalArgumentException e) {
+      throw new ServiceCombRuntimeException("value type not support");
+    }
+    Properties properties = new Properties();
+    Map<String, String> kvMap = new HashMap<>();
+    try {
+      if (vtype == (ValueType.yaml) || vtype == (ValueType.yml)) {
+        YamlPropertiesFactoryBean yamlFactory = new YamlPropertiesFactoryBean();
+        yamlFactory.setResources(new ByteArrayResource(kvDoc.getValue().getBytes()));
+        properties = yamlFactory.getObject();
+      } else if (vtype == (ValueType.properties)) {
+        properties.load(new StringReader(kvDoc.getValue()));
+      } else if (vtype == (ValueType.text) || vtype == (ValueType.string)) {
+        kvMap.put(kvDoc.getKey(), kvDoc.getValue());
+        return kvMap;
+      } else {
+        kvMap.put(kvDoc.getKey(), kvDoc.getValue());
+        return kvMap;
+      }
+      kvMap = toMap(kvDoc.getKey(), properties);
+      return kvMap;
```

Comment:
This code is very complicated, you can use some desgin patterns to make it simpler, such Factory based on valueType.

Reply:
using switch to simplify